modules/table/src/test/java/org/apache/ignite/internal/table/distributed/replication/PartitionReplicaListenerTest.java

```diff
@@ -2748,4 +2780,41 @@ private interface RwListenerInvocation {
     private interface RoListenerInvocation {
         CompletableFuture<?> invoke(UUID targetTxId, HybridTimestamp readTimestamp, TestKey key);
     }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void testStaleTxOperationAfterIndexStartBuilding(boolean readOnly) {
+        fireHashIndexStartBuildingEventForStaleTxOperation();
+
+        UUID txId = newTxId();
+        long beginTs = beginTimestamp(txId).longValue();
+
+        when(catalogService.activeCatalogVersion(eq(beginTs))).thenReturn(0);
```

Comment:
Is 0 a valid catalog version? Doesn't it start with 1?

Reply:
t starts at 0, you can look here: `org.apache.ignite.internal.catalog.CatalogManagerImpl#start`